ios/Classes/RudderSdkFlutterPlugin.m

```diff
@@ -109,16 +109,12 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
     else if ([@"putDeviceToken" isEqualToString:call.method]) {
         if([call.arguments objectForKey:@"deviceToken"]) {
             NSString* token =  [call.arguments objectForKey:@"deviceToken"];
-            if ([RSClient sharedInstance] == nil) {
-                return;
-            }
-            RSContext* rudderContext = [[RSClient sharedInstance] getContext];
-            if (rudderContext != nil && [token length] != 0) {
-                [rudderContext putDeviceToken:token];
+            if ([token length] != 0) {
+                [RSClient putDeviceToken:token];
             }
         }
         return;
-    } else if ([@"setAdvertisingId" isEqualToString:call.method]) {
+    } else if ([@"putAdvertisingId" isEqualToString:call.method]) {
```

Comment:
The checking should be other way around. :D 
`([call.method isEqualToString:@"putAdvertisingId"])`

Reply:
Updated